### ISSUE TITLE
No Detection on testing custom train model 

### ISSUE THREAD

**Issue by NaeemKhan333** [open-mmlab/mmdetection#4274]
I have trained the cascade_mask_rcnn_hrnetv2p_w32_20e_coco model on custom train model for three classes.I have used coco json format for training of model. it is giving me following log during the training

2020-12-09 15:00:00,430 - mmdet - INFO - Start running, host: root@290305e98ef9, work_dir: /content/drive/My Drive/mmdetection_object_detection_demo/mmdetection/work_dirs/cascade_mask_rcnn_hrnetv2p_w32_20e_coco
2020-12-09 15:00:00,431 - mmdet - INFO - workflow: [('train', 1)], max: 100 epochs
/usr/local/lib/python3.6/dist-packages/torch/nn/functional.py:2973: UserWarning: Default upsampling behavior when mode=bilinear is changed to align_corners=False since 0.4.0. Please specify align_corners=True if the old behavior is desired. See the documentation of nn.Upsample for details.
  "See the documentation of nn.Upsample for details.".format(mode))
2020-12-09 15:01:04,808 - mmdet - INFO - Epoch [1][50/770]	lr: 1.978e-03, eta: 1 day, 3:30:34, time: 1.287, data_time: 0.073, memory: 9197, loss_rpn_cls: 0.4559, loss_rpn_bbox: 0.0851, s0.loss_cls: 0.6585, s0.acc: 88.9141, s0.loss_bbox: 0.0368, s0.loss_mask: 0.3046, s1.loss_cls: 0.3321, s1.acc: 93.9238, s1.loss_bbox: 0.0108, s1.loss_mask: 0.1793, s2.loss_cls: 0.1782, s2.acc: 98.6797, s2.loss_bbox: 0.0021, s2.loss_mask: 0.1850, loss: 2.4285
2020-12-09 15:02:12,257 - mmdet - INFO - Epoch [1][100/770]	lr: 3.976e-03, eta: 1 day, 4:09:13, time: 1.349, data_time: 0.017, memory: 9286, loss_rpn_cls: 0.0841, loss_rpn_bbox: 0.0599, s0.loss_cls: 0.4245, s0.acc: 96.8809, s0.loss_bbox: 0.0722, s0.loss_mask: 0.2387, s1.loss_cls: 0.2213, s1.acc: 97.8672, s1.loss_bbox: 0.0354, s1.loss_mask: 0.0971, s2.loss_cls: 0.1107, s2.acc: 98.7051, s2.loss_bbox: 0.0085, s2.loss_mask: 0.0315, loss: 1.3839
2020-12-09 15:03:18,895 - mmdet - INFO - Epoch [1][150/770]	lr: 5.974e-03, eta: 1 day, 4:14:25, time: 1.333, data_time: 0.018, memory: 9402, loss_rpn_cls: 0.0619, loss_rpn_bbox: 0.0612, s0.loss_cls: 0.3015, s0.acc: 96.5352, s0.loss_bbox: 0.0915, s0.loss_mask: 0.2569, s1.loss_cls: 0.1794, s1.acc: 97.3066, s1.loss_bbox: 0.0474, s1.loss_mask: 0.1024, s2.loss_cls: 0.0965, s2.acc: 98.3594, s2.loss_bbox: 0.0118, s2.loss_mask: 0.0354, loss: 1.2458
2020-12-09 15:04:26,702 - mmdet - INFO - Epoch [1][200/770]	lr: 7.972e-03, eta: 1 day, 4:23:57, time: 1.356, data_time: 0.021, memory: 9402, loss_rpn_cls: 0.0832, loss_rpn_bbox: 0.0760, s0.loss_cls: 0.2321, s0.acc: 95.9004, s0.loss_bbox: 0.1083, s0.loss_mask: 0.2704, s1.loss_cls: 0.1438, s1.acc: 96.6426, s1.loss_bbox: 0.0534, s1.loss_mask: 0.1017, s2.loss_cls: 0.0828, s2.acc: 97.8359, s2.loss_bbox: 0.0148, s2.loss_mask: 0.0356, loss: 1.2021
2020-12-09 15:05:30,204 - mmdet - INFO - Epoch [1][250/770]	lr: 9.970e-03, eta: 1 day, 4:07:11, time: 1.270, data_time: 0.019, memory: 9402, loss_rpn_cls: 0.3703, loss_rpn_bbox: 0.8771, s0.loss_cls: nan, s0.acc: 56.1836, s0.loss_bbox: nan, s0.loss_mask: 23.1429, s1.loss_cls: nan, s1.acc: 76.8734, s1.loss_bbox: nan, s1.loss_mask: 1.5815, s2.loss_cls: nan, s2.acc: 77.4301, s2.loss_bbox: nan, s2.loss_mask: 0.0615, loss: nan
2020-12-09 15:06:26,744 - mmdet - INFO - Epoch [1][300/770]	lr: 1.197e-02, eta: 1 day, 3:25:59, time: 1.131, data_time: 0.017, memory: 9402, loss_rpn_cls: 0.5307, loss_rpn_bbox: 0.0789, s0.loss_cls: nan, s0.acc: 0.3203, s0.loss_bbox: nan, s0.loss_mask: 25.6634, s1.loss_cls: nan, s1.acc: 49.8946, s1.loss_bbox: nan, s1.loss_mask: 0.0110, s2.loss_cls: nan, s2.acc: 49.8946, s2.loss_bbox: nan, s2.loss_mask: 0.0060, loss: nan
2020-12-09 15:07:21,946 - mmdet - INFO - Epoch [1][350/770]	lr: 1.397e-02, eta: 1 day, 2:51:25, time: 1.104, data_time: 0.018, memory: 9402, loss_rpn_cls: 0.4244, loss_rpn_bbox: 0.0825, s0.loss_cls: nan, s0.acc: 1.0703, s0.loss_bbox: nan, s0.loss_mask: 0.1908, s1.loss_cls: nan, s1.acc: 49.8663, s1.loss_bbox: nan, s1.loss_mask: 0.0073, s2.loss_cls: nan, s2.acc: 49.8663, s2.loss_bbox: nan, s2.loss_mask: 0.0059, loss: nan
2020-12-09 15:08:18,355 - mmdet - INFO - Epoch [1][400/770]	lr: 1.596e-02, eta: 1 day, 2:29:06, time: 1.128, data_time: 0.019, memory: 9402, loss_rpn_cls: 0.3583, loss_rpn_bbox: 0.0680, s0.loss_cls: nan, s0.acc: 1.3047, s0.loss_bbox: nan, s0.loss_mask: 0.2160, s1.loss_cls: nan, s1.acc: 48.2250, s1.loss_bbox: nan, s1.loss_mask: 0.0057, s2.loss_cls: nan, s2.acc: 48.2250, s2.loss_bbox: nan, s2.loss_mask: 0.0045, loss: nan
2020-12-09 15:09:14,494 - mmdet - INFO - Epoch [1][450/770]	lr: 1.796e-02, eta: 1 day, 2:10:47, time: 1.123, data_time: 0.018, memory: 9402, loss_rpn_cls: 0.3456, loss_rpn_bbox: 0.0721, s0.loss_cls: nan, s0.acc: 1.4453, s0.loss_bbox: nan, s0.loss_mask: 0.2611, s1.loss_cls: nan, s1.acc: 49.1674, s1.loss_bbox: nan, s1.loss_mask: 0.0059, s2.loss_cls: nan, s2.acc: 49.1674, s2.loss_bbox: nan, s2.loss_mask: 0.0044, loss: nan
2020-12-09 15:10:12,922 - mmdet - INFO - Epoch [1][500/770]	lr: 1.996e-02, eta: 1 day, 2:01:46, time: 1.169, data_time: 0.019, memory: 9402, loss_rpn_cls: 0.2846, loss_rpn_bbox: 0.0540, s0.loss_cls: nan, s0.acc: 2.0352, s0.loss_bbox: nan, s0.loss_mask: 0.2760, s1.loss_cls: nan, s1.acc: 49.0887, s1.loss_bbox: nan, s1.loss_mask: 0.0046, s2.loss_cls: nan, s2.acc: 49.0887, s2.loss_bbox: nan, s2.loss_mask: 0.0032, loss: nan
2020-12-09 15:11:10,844 - mmdet - INFO - Epoch [1][550/770]	lr: 2.000e-02, eta: 1 day, 1:53:03, time: 1.158, data_time: 0.016, memory: 9402, loss_rpn_cls: 0.3248, loss_rpn_bbox: 0.0693, s0.loss_cls: nan, s0.acc: 1.2578, s0.loss_bbox: nan, s0.loss_mask: 0.2344, s1.loss_cls: nan, s1.acc: 48.4576, s1.loss_bbox: nan, s1.loss_mask: 0.0032, s2.loss_cls: nan, s2.acc: 48.4576, s2.loss_bbox: nan, s2.loss_mask: 0.0020, loss: nan
2020-12-09 15:12:07,954 - mmdet - INFO - Epoch [1][600/770]	lr: 2.000e-02, eta: 1 day, 1:43:53, time: 1.142, data_time: 0.020, memory: 9402, loss_rpn_cls: 0.3204, loss_rpn_bbox: 0.0704, s0.loss_cls: nan, s0.acc: 2.0801, s0.loss_bbox: nan, s0.loss_mask: 0.2596, s1.loss_cls: nan, s1.acc: 50.0763, s1.loss_bbox: nan, s1.loss_mask: 0.0042, s2.loss_cls: nan, s2.acc: 50.0763, s2.loss_bbox: nan, s2.loss_mask: 0.0028, loss: nan
2020-12-09 15:13:04,250 - mmdet - INFO - Epoch [1][650/770]	lr: 2.000e-02, eta: 1 day, 1:34:24, time: 1.126, data_time: 0.017, memory: 9402, loss_rpn_cls: 0.3282, loss_rpn_bbox: 0.0708, s0.loss_cls: nan, s0.acc: 1.4551, s0.loss_bbox: nan, s0.loss_mask: 0.2447, s1.loss_cls: nan, s1.acc: 48.1672, s1.loss_bbox: nan, s1.loss_mask: 0.0039, s2.loss_cls: nan, s2.acc: 48.1672, s2.loss_bbox: nan, s2.loss_mask: 0.0023, loss: nan
2020-12-09 15:14:00,980 - mmdet - INFO - Epoch [1][700/770]	lr: 2.000e-02, eta: 1 day, 1:26:56, time: 1.135, data_time: 0.016, memory: 9402, loss_rpn_cls: 0.2958, loss_rpn_bbox: 0.0585, s0.loss_cls: nan, s0.acc: 1.6367, s0.loss_bbox: nan, s0.loss_mask: 0.2646, s1.loss_cls: nan, s1.acc: 48.1508, s1.loss_bbox: nan, s1.loss_mask: 0.0019, s2.loss_cls: nan, s2.acc: 48.1508, s2.loss_bbox: nan, s2.loss_mask: 0.0011, loss: nan
2020-12-09 15:14:57,781 - mmdet - INFO - Epoch [1][750/770]	lr: 2.000e-02, eta: 1 day, 1:20:27, time: 1.136, data_time: 0.017, memory: 9402, loss_rpn_cls: 0.3063, loss_rpn_bbox: 0.0687, s0.loss_cls: nan, s0.acc: 1.7285, s0.loss_bbox: nan, s0.loss_mask: 0.2867, s1.loss_cls: nan, s1.acc: 48.5758, s1.loss_bbox: nan, s1.loss_mask: 0.0033, s2.loss_cls: nan, s2.acc: 48.5758, s2.loss_bbox: nan, s2.loss_mask: 0.0020, loss: nan
2020-12-09 15:15:20,093 - mmdet - INFO - Saving checkpoint at 1 epochs
[>>] 1556/1556, 6.3 task/s, elapsed: 247s, ETA:     0s2020-12-09 15:19:29,783 - mmdet - INFO - Evaluating bbox...
Loading and preparing results...
2020-12-09 15:19:29,784 - mmdet - ERROR - The testing results of the whole dataset is empty.
2020-12-09 15:19:29,786 - mmdet - INFO - Exp name: cascade_mask_rcnn_hrnetv2p_w32_20e_coco.py
2020-12-09 15:19:29,786 - mmdet - INFO - Epoch(val) [1][770]	
2020-12-09 15:20:31,019 - mmdet - INFO - Epoch [2][50/770]	lr: 2.000e-02, eta: 1 day, 0:44:09, time: 1.224, data_time: 0.063, memory: 9402, loss_rpn_cls: 0.2979, loss_rpn_bbox: 0.0625, s0.loss_cls: nan, s0.acc: 1.4727, s0.loss_bbox: nan, s0.loss_mask: 0.2738, s1.loss_cls: nan, s1.acc: 48.6368, s1.loss_bbox: nan, s1.loss_mask: 0.0042, s2.loss_cls: nan, s2.acc: 48.6368, s2.loss_bbox: nan, s2.loss_mask: 0.0025, loss: nan
2020-12-09 15:21:28,046 - mmdet - INFO - Epoch [2][100/770]	lr: 2.000e-02, eta: 1 day, 0:41:06, time: 1.141, data_time: 0.017, memory: 9402, loss_rpn_cls: 0.3007, loss_rpn_bbox: 0.0586, s0.loss_cls: nan, s0.acc: 1.9902, s0.loss_bbox: nan, s0.loss_mask: 0.2526, s1.loss_cls: nan, s1.acc: 48.6786, s1.loss_bbox: nan, s1.loss_mask: 0.0013, s2.loss_cls: nan, s2.acc: 48.6786, s2.loss_bbox: nan, s2.loss_mask: 0.0007, loss: nan
2020-12-09 15:22:25,833 - mmdet - INFO - Epoch [2][150/770]	lr: 2.000e-02, eta: 1 day, 0:39:20, time: 1.156, data_time: 0.016, memory: 9402, loss_rpn_cls: 0.3119, loss_rpn_bbox: 0.0664, s0.loss_cls: nan, s0.acc: 1.8926, s0.loss_bbox: nan, s0.loss_mask: 0.2522, s1.loss_cls: nan, s1.acc: 48.4246, s1.loss_bbox: nan, s1.loss_mask: 0.0027, s2.loss_cls: nan, s2.acc: 48.4246, s2.loss_bbox: nan, s2.loss_mask: 0.0014, loss: nan
2020-12-09 15:23:24,674 - mmdet - INFO - Epoch [2][200/770]	lr: 2.000e-02, eta: 1 day, 0:39:01, time: 1.177, data_time: 0.023, memory: 9402, loss_rpn_cls: 0.3139, loss_rpn_bbox: 0.0598, s0.loss_cls: nan, s0.acc: 2.6074, s0.loss_bbox: nan, s0.loss_mask: 0.2666, s1.loss_cls: nan, s1.acc: 49.2508, s1.loss_bbox: nan, s1.loss_mask: 0.0044, s2.loss_cls: nan, s2.acc: 49.2508, s2.loss_bbox: nan, s2.loss_mask: 0.0025, loss: nan
2020-12-09 15:24:22,291 - mmdet - INFO - Epoch [2][250/770]	lr: 2.000e-02, eta: 1 day, 0:37:08, time: 1.152, data_time: 0.020, memory: 9402, loss_rpn_cls: 0.3193, loss_rpn_bbox: 0.0767, s0.loss_cls: nan, s0.acc: 2.1250, s0.loss_bbox: nan, s0.loss_mask: 0.2536, s1.loss_cls: nan, s1.acc: 48.9561, s1.loss_bbox: nan, s1.loss_mask: 0.0057, s2.loss_cls: nan, s2.acc: 48.9561, s2.loss_bbox: nan, s2.loss_mask: 0.0028, loss: nan
2020-12-09 15:25:19,096 - mmdet - INFO - Epoch [2][300/770]	lr: 2.000e-02, eta: 1 day, 0:34:22, time: 1.136, data_time: 0.018, memory: 9402, loss_rpn_cls: 0.3028, loss_rpn_bbox: 0.0567, s0.loss_cls: nan, s0.acc: 1.8555, s0.loss_bbox: nan, s0.loss_mask: 0.2635, s1.loss_cls: nan, s1.acc: 47.3700, s1.loss_bbox: nan, s1.loss_mask: 0.0030, s2.loss_cls: nan, s2.acc: 47.3700, s2.loss_bbox: nan, s2.loss_mask: 0.0015, loss: nan
2020-12-09 15:26:13,842 - mmdet - INFO - Epoch [2][350/770]	lr: 2.000e-02, eta: 1 day, 0:29:26, time: 1.095, data_time: 0.016, memory: 9402, loss_rpn_cls: 0.3124, loss_rpn_bbox: 0.0691, s0.loss_cls: nan, s0.acc: 1.5820, s0.loss_bbox: nan, s0.loss_mask: 0.2379, s1.loss_cls: nan, s1.acc: 49.1186, s1.loss_bbox: nan, s1.loss_mask: 0.0038, s2.loss_cls: nan, s2.acc: 49.1186, s2.loss_bbox: nan, s2.loss_mask: 0.0019, loss: nan
2020-12-09 15:27:08,085 - mmdet - INFO - Epoch [2][400/770]	lr: 2.000e-02, eta: 1 day, 0:24:18, time: 1.085, data_time: 0.016, memory: 9402, loss_rpn_cls: 0.3009, loss_rpn_bbox: 0.0595, s0.loss_cls: nan, s0.acc: 1.6680, s0.loss_bbox: nan, s0.loss_mask: 0.2531, s1.loss_cls: nan, s1.acc: 49.9116, s1.loss_bbox: nan, s1.loss_mask: 0.0052, s2.loss_cls: nan, s2.acc: 49.9116, s2.loss_bbox: nan, s2.loss_mask: 0.0023, loss: nan
2020-12-09 15:28:01,640 - mmdet - INFO - Epoch [2][450/770]	lr: 2.000e-02, eta: 1 day, 0:18:48, time: 1.071, data_time: 0.018, memory: 9402, loss_rpn_cls: 0.2758, loss_rpn_bbox: 0.0565, s0.loss_cls: nan, s0.acc: 1.8613, s0.loss_bbox: nan, s0.loss_mask: 0.2599, s1.loss_cls: nan, s1.acc: 48.8371, s1.loss_bbox: nan, s1.loss_mask: 0.0075, s2.loss_cls: nan, s2.acc: 48.8371, s2.loss_bbox: nan, s2.loss_mask: 0.0031, loss: nan
2020-12-09 15:28:58,841 - mmdet - INFO - Epoch [2][500/770]	lr: 2.000e-02, eta: 1 day, 0:17:17, time: 1.144, data_time: 0.018, memory: 9402, loss_rpn_cls: 0.3046, loss_rpn_bbox: 0.0683, s0.loss_cls: nan, s0.acc: 1.7148, s0.loss_bbox: nan, s0.loss_mask: 0.2602, s1.loss_cls: nan, s1.acc: 48.0007, s1.loss_bbox: nan, s1.loss_mask: 0.0042, s2.loss_cls: nan, s2.acc: 48.0007, s2.loss_bbox: nan, s2.loss_mask: 0.0017, loss: nan

After Completing the training, I have done inferences on the saved checkpoints.It is giving me empty result, no detection found on the training data as well.Can anybody guide me what could be the reason.Thanks

**Comment by ZwwWayne** [comment]
This is because the loss is Nan and the model training is failed.

**Comment by wyh196646** [comment]
This is really a efficient sloution

**Comment by wyh196646** [comment]
This is really a efficient soloution
